metrics/src/main/scala/org/broadinstitute/dsde/rawls/metrics/InstrumentedRetry.scala

```diff
@@ -0,0 +1,30 @@
+package org.broadinstitute.dsde.rawls.metrics
+
+import com.typesafe.scalalogging.LazyLogging
+import nl.grons.metrics.scala.Histogram
+import org.broadinstitute.dsde.rawls.util.Retry
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Created by rtitle on 8/3/17.
+  */
+trait InstrumentedRetry extends Retry {
+  this: WorkbenchInstrumented with LazyLogging =>
+
+  /**
+    * Converts an AccumulatingFuture[A] to a Future[A].
+    * Given an implicit Histogram, instruments the number of failures in the histogram.
+    */
+  protected implicit def accumulatingFutureToFuture[A](af: AccumulatingFuture[A])(implicit histo: Histogram, executionContext: ExecutionContext): Future[A] = {
```

Comment:
Yeah it's not overridable because of the implicit parameter.

This might work:
```
    val instrumentedAccumulatingFuture = af.andThen {
      case Success(Left(errors)) =>
        histo += errors.toList.size
      case Success(Right((errors, a))) =>
        histo += errors.size
    }
    super.accumulatingFutureToFuture(instrumentedAccumulatingFuture)
```